crates/red_knot_python_semantic/src/types/call/bind.rs

```diff
@@ -1173,6 +1322,21 @@ impl<'db> BindingError<'db> {
                     ),
                 );
             }
+
+            Self::InternalCallError(reason) => {
+                context.report_lint(
+                    &CALL_NON_CALLABLE,
+                    Self::get_node(node, None),
+                    format_args!(
+                        "Internal error while evaluating call{}: {reason}",
```

Comment:
I think we might want to avoid the "Internal error" language in user-facing messages, as I suspect users may interpret it as "there's a bug in red-knot". Maybe just "Call{} failed: {reason}"?